crates/bevy_pbr/src/render/gpu_preprocess.rs

```diff
@@ -57,20 +64,44 @@ pub struct GpuMeshPreprocessPlugin {
 
 /// The render node for the mesh uniform building pass.
 pub struct GpuPreprocessNode {
-    view_query: QueryState<(Entity, Read<PreprocessBindGroup>)>,
+    view_query: QueryState<(
+        Entity,
+        Read<PreprocessBindGroup>,
+        Read<ViewUniformOffset>,
+        Has<GpuCulling>,
+    )>,
 }
 
-/// The compute shader pipeline for the mesh uniform building pass.
+/// The compute shader pipelines for the mesh uniform building pass.
 #[derive(Resource)]
+pub struct PreprocessPipelines {
+    /// The pipeline used for CPU culling. This pipeline doesn't populate
+    /// indirect parameters.
+    pub direct: PreprocessPipeline,
+    /// The pipeline used for GPU culling. This pipeline populates indirect
+    /// parameters.
+    pub gpu_culling: PreprocessPipeline,
+}
+
+/// The pipeline for the GPU mesh preprocessing shader.
 pub struct PreprocessPipeline {
-    /// The single bind group layout for the compute shader.
+    /// The bind group layout for the compute shader.
     pub bind_group_layout: BindGroupLayout,
     /// The pipeline ID for the compute shader.
     ///
-    /// This gets filled in in `prepare_preprocess_pipeline`.
+    /// This gets filled in in `prepare_preprocess_pipelines`.
     pub pipeline_id: Option<CachedComputePipelineId>,
 }
 
+/// Specifies variants of the mesh preprocessing shader.
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub struct PreprocessPipelineKey {
+    /// Whether GPU culling is in use.
+    ///
+    /// This `#define`'s `GPU_CULLING` in the shader.
+    pub gpu_culling: bool,
```

Comment:
Is it worth making this is bitflags over a u8 so that if/when it grows, it grows using bitflags and not with additional members?

Reply:
Done.